### DISCUSSION TITLE
Problem with Esapi Functions

### DISCUSSION THREAD

**Discussion by mike2907** [MasaCMS/MasaCMS#169] [General]
Hi Folks,

I'm trying to move from Mura CMS 7.1 to Masa CMS. I've tried upgrading and I've tried a clean install. I'm running into a problem with a function that's being declared in Masa CMS. I'm running the latest release of Lucee on a Windows 2016 server. Here's the error message I'm receiving.
 
```
Lucee 5.3.10.120 Error (template)

Message	The name [esapiEncode] is already used by a built in Function

Stacktrace	The Error Occurred in
C:\inetpub\MasaCMS\core\mura\backport\esapiencode.cfm: line 1

1: <cffunction name="esapiEncode" output="false">

2: <cfargument name="encodeFor" type="string" default="html" hint="encode for what, valid values are: - css: for output inside Cascading Style Sheets (CSS) - dn: for output in LDAP Distinguished Names - html: for output inside HTML - html_attr: for output inside HTML Attributes - javascript: for output inside Javascript - ldap: for output in LDAP queries - url: for output in URL - vbscript: for output inside vbscript - xml: for output inside XML - xml_attr: for output inside XML Attributes - xpath: for output in XPath">

3: <cfargument name="inputString" type="string" required="true" hint="Required. String to encode">

called from C:\inetpub\MasaCMS\core\mura\backport\backport.cfm: line 22

called from C:\inetpub\MasaCMS\admin\Application.cfc: line 83

called from C:\inetpub\MasaCMS\admin\Application.cfc: line 76
```

Is there a workaround for this? Thanks,

Mike

**Comment by mike2907** [comment]
I solved this by deleting the OWASP Esapi extension from Lucee, setting up the sites, and then reinstalling the extension.

**Reply by tomchiverton** [reply]
There's also a Lucee bug if you upgraded from older versions, `rm tomcat/lucee-server/bundles/org-lucee-esapi-2-2-0-0.jar` and restart (they'll be a newer version of the jar with dots)

**Comment by mike2907** [comment]
I ended up removing the Esepi module completely and then installing the latest version. Should I leave things as is, or should I remove it and roll back? Thanks,

Mike

**Comment by tomchiverton** [comment]
I removed the "spare" .jar - makes sense to tidy it up.
